pkg/config/namewatcher.go

```diff
@@ -0,0 +1,83 @@
+package config
+
+import (
+	"os"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+)
+
+const (
+	sharedSecretReservedNamesEnvVarName    = "RESERVED_SHARED_SECRET_NAMES"
+	sharedConfigMapReservedNamesEnvVarName = "RESERVED_SHARED_CONFIGMAP_NAMES"
+)
+
+type ReservedNames struct {
+	currentSecretNames    map[string]types.NamespacedName
+	currentConfigMapNames map[string]types.NamespacedName
+}
+
+func SetupNameReservation() *ReservedNames {
+	rn := &ReservedNames{}
+	sharedSecretReservations := os.Getenv(sharedSecretReservedNamesEnvVarName)
+	rn.currentSecretNames = parseEnvVar(sharedSecretReservations, sharedSecretReservedNamesEnvVarName)
+	klog.Infof("reserved shared secret env %s and map %#v", sharedSecretReservations, rn.currentSecretNames)
+	sharedConfigMapReservations := os.Getenv(sharedConfigMapReservedNamesEnvVarName)
+	rn.currentConfigMapNames = parseEnvVar(sharedConfigMapReservations, sharedConfigMapReservedNamesEnvVarName)
+	klog.Infof("reserved shared configmap env %s and map %#v", sharedConfigMapReservations, rn.currentConfigMapNames)
+	return rn
+}
+
+func parseEnvVar(val, envvarName string) map[string]types.NamespacedName {
+	ret := map[string]types.NamespacedName{}
+	entries := strings.Split(val, ";")
+	for _, entry := range entries {
+		a := strings.Split(entry, ":")
+		if len(a) < 3 {
+			if len(entry) > 0 {
+				klog.Warningf("env var %s has bad entry %s", envvarName, entry)
+			}
+			continue
+		}
+		nsName := types.NamespacedName{
+			Namespace: strings.TrimSpace(a[1]),
+			Name:      strings.TrimSpace(a[2]),
+		}
```

Comment:
Interesting idea, but I don't think that buys us much of anything.  No attempts at creation are done with those values.
They either match actual shares and their references or they don't.
I also don't think seeing if such `Secrets` or `ConfigMaps` exist makes sense here.  We want to allow for creation of the shared by other openshift components, or installed operators, which could occur after we start up.  Finally, one of the motivators of moving from configmap to envvar was to avoid making any api calls from the webhook.

So punting on this one.

Reply:
I will defer to your decision based on your explanation. Thanks